src/main/java/com/peerrequest/app/api/DirectRequestsController.java

```diff
@@ -0,0 +1,264 @@
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ * The RequestsController handles all directRequests interaction with the backend.
+ */
+@RestController
+@ApiControllerPrefix
+public class DirectRequestsController extends ServiceBasedController {
+
+    private final int maxPageSize = 100;
+
+    /**
+     * Gets a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     *
+     * @return directRequestProcess of entry
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
+    public DirectRequestProcess.Dto getDirectRequestProcess(@PathVariable("entry_id") final Long entryId) {
+        var option = this.directRequestProcessService.get(entryId);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request process does not exist");
+        }
+
+        return option.get().toDto();
+    }
+
+    /**
+     * Creates a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     * @param dto containing the open slots
+     *
+     * @return the added directRequestProcess
+     */
+    @PostMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
+    public DirectRequestProcess.Dto createDirectRequestProcess(@PathVariable("entry_id") Long entryId,
+                                                        @RequestBody DirectRequestProcess.Dto dto,
+                                                        @AuthenticationPrincipal OAuth2User user) {
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        if (dto.openSlots().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "open slots is required");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+
+        if (!user.getAttribute("sub").toString().equals(researcherId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may create a request process");
+        }
+        // TODO: check if there is an existent Bidding process with the entry
+        var directRequestProcessDto = DirectRequestProcess.fromDto(dto, entryId);
+        return this.directRequestProcessService.create(directRequestProcessDto.toDto()).toDto();
+    }
+
+    /**
+     * Patches a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     * @param dto updater containing the amount of openSlots (int)
+     *
+     * @return updated directRequestProcess
+     */
+    @PutMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
```

Comment:
remove:  produce = "aplication/json"
to match the other controllers

Reply:
Done.